src/main/java/org/candlepin/subscriptions/security/OptInController.java

```diff
@@ -51,128 +47,69 @@
 public class OptInController {
   private static final Logger log = LoggerFactory.getLogger(OptInController.class);
 
-  private AccountConfigRepository accountConfigRepository;
   private OrgConfigRepository orgConfigRepository;
   private ApplicationClock clock;
-  private AccountService accountService;
 
   @Autowired
-  public OptInController(
-      ApplicationClock clock,
-      AccountConfigRepository accountConfigRepo,
-      OrgConfigRepository orgConfigRepo,
-      AccountService accountService) {
+  public OptInController(ApplicationClock clock, OrgConfigRepository orgConfigRepo) {
     this.clock = clock;
-    this.accountConfigRepository = accountConfigRepo;
     this.orgConfigRepository = orgConfigRepo;
-    this.accountService = accountService;
   }
 
   // Separate isolated transaction needed in order to prevent opt-in errors rolling back metrics
   // updates
   @Transactional(propagation = Propagation.REQUIRES_NEW)
-  public OptInConfig optIn(String accountNumber, String orgId, OptInType optInType) {
-    return performOptIn(accountNumber, orgId, optInType);
+  public OptInConfig optIn(String orgId, OptInType optInType) {
+    return performOptIn(orgId, optInType);
   }
 
-  private OptInConfig performOptIn(String accountNumber, String orgId, OptInType optInType) {
+  private OptInConfig performOptIn(String orgId, OptInType optInType) {
     OffsetDateTime now = clock.now();
-
-    Optional<AccountConfig> accountData =
-        accountConfigRepository.createOrUpdateAccountConfig(accountNumber, orgId, now, optInType);
     Optional<OrgConfig> orgData =
         orgConfigRepository.createOrUpdateOrgConfig(orgId, now, optInType);
-    return buildDto(
-        buildMeta(accountNumber, orgId),
-        buildOptInAccountDTO(accountData),
-        buildOptInOrgDTO(orgData));
-  }
-
-  // Separate isolated transaction needed in order to prevent opt-in errors rolling back metrics
-  // updates
-  @Transactional(propagation = Propagation.REQUIRES_NEW)
-  public void optInByAccountNumber(String accountNumber, OptInType optInType) {
-    if (accountConfigRepository.existsByAccountNumber(accountNumber)) {
-      return;
-    }
-    String orgId = accountService.lookupOrgId(accountNumber);
-    log.info("Opting in account/orgId: {}/{}", accountNumber, orgId);
-    performOptIn(accountNumber, orgId, optInType);
+    return buildDto(buildMeta(orgId), buildOptInOrgDTO(orgData));
   }
 
   // Separate isolated transaction needed in order to prevent opt-in errors rolling back metrics
   // updates
   @Transactional(propagation = Propagation.REQUIRES_NEW)
   public void optInByOrgId(String orgId, OptInType optInType) {
-    if (!accountConfigRepository.existsById(orgId)) {
+    if (Boolean.FALSE.equals(orgConfigRepository.existsByOrgId(orgId))) {
```

Comment:
Using `Boolean.FALSE.equals(` is no longer necessary because existsByOrgId now returns the primitive boolean.